csrc/flash_attn/src/fmha_fprop_kernel_1xN.h

```diff
@@ -678,6 +678,8 @@ inline __device__ void device_1xN_loop(const Params &params) {
     // the attention matrix. This way, as long as we have the batch, head, and the location of
     // the 16 x 16 block within the attention matrix, we can generate the exact same dropout pattern.
     auto seeds = at::cuda::philox::unpack(params.philox_args);
+    params.rng_state[0] = std::get<0>(seeds);
```

Comment:
Should only 1 thread write this to global memory (e.g thread 0 of block (0, 0, 0))?

Reply:
You're right that would be better. Made the changes and retested!